proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,800 @@
+    **MomentumBalanceType.momentumTotal** - single momentum balance for material,
+    **MomentumBalanceType.momentumPhase** - momentum balances for each phase.}"""))
+
+    CONFIG.declare("finite_elements", ConfigValue(
+            default=20,
+            domain=int,
+            description="Number of finite elements in length domain",
+            doc="""Number of finite elements to use when discretizing length 
+            domain (default=20)"""))
+
+    CONFIG.declare("collocation_points", ConfigValue(
+            default=5,
+            domain=int,
+            description="Number of collocation points per finite element",
+            doc="""Number of collocation points to use per finite element when
+            discretizing length domain (default=5)"""))
+
+    def _process_config(self):
+        #TODO: add config errors here:
+        # prevent permeate_side.deltaP
+        for c in self.config.property_package.component_list:
+            comp = self.config.property_package.get_component(c)
+            try:
+                if comp.is_solvent():
+                    self.solvent_list.add(c)
+                if comp.is_solute():
+                    self.solute_list.add(c)
+            except TypeError:
+                raise ConfigurationError("RO model only supports one solvent and one or more solutes,"
+                                         "the provided property package has specified a component '{}' "
+                                         "that is not a solvent or solute".format(c))
+        if len(self.solvent_list) > 1:
+            raise ConfigurationError("RO model only supports one solvent component,"
+                                     "the provided property package has specified {} solvent components"
+                                     .format(len(self.solvent_list)))
+
+        if self.config.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the "
+                "reverse osmosis module. "
+                "Defaulting to finite "
+                "difference method."
+            )
+            self.config.transformation_method = "dae.finite_difference"
+
+        if self.config.transformation_scheme is useDefault:
+            _log.warning(
+                "Discretization scheme was "
+                "not specified for the "
+                "reverse osmosis module."
+                "Defaulting to backward finite "
+                "difference."
+            )
+            self.config.transformation_scheme = "BACKWARD"
+
+
+    def build(self):
+        """
+        Build 1D RO model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.solvent_list = Set()
```

Comment:
You should be able to get these from the property package, e.g. `self.config.property_package.solvent_set`

Reply:
I think that `solvent_set` and `solute_set` weren't defined in the NaCl property model, so would have to modify the property model for that to work.

Reply:
That is true. That said, from the IDAES standards perspective at least, it is the property package's job to define these sorts of things.

Reply:
Yes the solvent_set and solute_set should be defined in the property package. It really should be automatically created from defining the components like the component_list and phase_list on IDAES-side. I was planning on making a PR for IDAES, but it got deprioritized.

Reply:
The generic property framework already does this, so it is only the custom property packages which lack these, Making a general PR would need to keep in mind how this would impact the generic framework.

Reply:
For now, I will leave this be until we move forward on a PR for the property model. Also want to note that we'd have to revise this segment of code in the 0D RO model as well.